internal/cli/cli.go

```diff
@@ -145,10 +162,14 @@ func (factory *commandFactory) Build(provider func() CommandDefinition) *cobra.C
 		Short: command.Description,
 		Long:  command.Help,
 		RunE: func(c *cobra.Command, a []string) error {
+			publicAPIKey := factory.profile.GetString(keyPublicAPIKey)
```

Comment:
I'm now thinking it might make sense to pass some extra details into the tracker constructor (after the `mode` arg, a `TrackerOptions` struct could work nicely here).  I think two right off the bat that would make sense (since they are constant throughout a command execution) would be this `PublicAPIKey` as well as a `primitive.NewObjectID().Hex()` (as the `executionID`).

that way, calls to `Track` can start to worry less about some of these other common details.  this alternate approach likely also forces you to tweak the `Track()` interface method a little so that information injected to the `tracker` (via its constructor) can be merged easily with the data being passed in at the time `Track` is called.  you definitely have a couple of way to approach this, feel free to take a stab or we can talk offline about some of them